### PR TITLE
KYP-626: Handle phone orders in magento plugin

### PR DESCRIPTION
**Pull request by nedick007**
Add phone email for customers with missing email. Discard sync if no email and phone number are present.

### CODE REVIEW

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo ready for review.

File: Helper/OrderSerializer.php [hunk 1]
```diff
@@ -31,15 +32,26 @@ public function serialize($order) {
...
             "countryCode"   => $orderBillingData->getCountryId(),
-            "phone"         => $orderBillingData->getTelephone(),
+            "phone"         => $orderPhone,
             "postcode"      => $orderBillingData->getPostcode(),
             "paymentMethod" => $order->getPayment()->getMethodInstance()->getTitle()
         ];
-    
+        
+        if (empty($order->getCustomerEmail())) {
+            if (isset($orderPhone)) {
```
Comment (evgeni-spasov-metrilo) [inline Helper/OrderSerializer.php]: `$orderPhone` will always be set, but it could be empty.

Comment (nedick007) [reply Helper/OrderSerializer.php]: Hmm, that is correct. Updated the logic.

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo bump

File: Helper/OrderSerializer.php [hunk 2]
```diff
@@ -31,15 +32,26 @@ public function serialize($order) {
...
+        
+        if (empty($order->getCustomerEmail())) {
+            if (empty($orderPhone)) {
+                return false;
+            } else {
+                $customerEmail = $orderPhone . '@phone_email';
+            }
+        } else {
+            $customerEmail = $order->getCustomerEmail();
+        }
```
Comment (evgeni-spasov-metrilo) [inline Helper/OrderSerializer.php]: Why not something simpler to read?
```php
if (!empty($order->getCustomerEmail())) {
  $customerEmail = $order->getCustomerEmail();
} elseif (!empty($orderPhone)) {
  $customerEmail = $orderPhone . '@phone_email';
} else {
  return false;
}
```

Comment (nedick007) [reply Helper/OrderSerializer.php]: Updating as you suggest.

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo bump

**Pull request merged**